client/src/main/java/se/fortnox/reactivewizard/client/RxClientProvider.java

```diff
@@ -54,11 +75,24 @@ public RxClientProvider(HttpClientConfig config, HealthRecorder healthRecorder)
             .pipelineConfigurator(UnsubscribeAwareHttpClientToConnectionBridge::configurePipeline);
 
         if (config.isHttps()) {
-            client = client.unsafeSecure();
+            return configureSsl(client, config.getHost(), config.getPort(), config.isUnsafeSecure());
         }
         return client;
     }
 
+    /**
+     * Allows for customization of the SSLEngine
+     *
+     * @return An configured SSLEngine
+     */
+    protected SSLEngine configureSslEngine(String host, int port) {
+        try {
+            return SSLContext.getDefault().createSSLEngine(host, port);
+        } catch (Throwable e) {
+            throw new RuntimeException("Unable to configure secure client", e);
```

Comment:
maybe "https client" or something? So when it appears in the logs, it will be easier to understand what kind of client we are talking about. Just a maybe.

Reply:
add test case for the exception maybe. The current testcases will not run through this code as i can see.

Reply:
createSSLEngine is public final which I'm having trouble mocking to throw an exception. I believe exceptions would be thrown here if the JRE doesnt have a default keystore or if commandline parameters for the keystore/truststore point to non-existant files.